src/test/java/CustomHandlerRouterTest.java

```diff
@@ -0,0 +1,40 @@
+import com.td.HttpServer.CustomHandlerRouter;
+import com.td.HttpServer.DefaultHandlerRouter;
+import com.td.HttpServer.HttpRequest;
+import com.td.HttpServer.IHandler;
+import com.td.HttpServer.HandlerGetDirectoryContents;
+import com.td.Mocks.MockHandler;
+import com.td.Mocks.MockFileIO;
+import java.util.HashMap;
+
+public class CustomHandlerRouterTest extends junit.framework.TestCase {
+
+  private DefaultHandlerRouter defaultHandlerRouter;
+  private CustomHandlerRouter customHandlerRouter;
+  private MockHandler mockHandler;
+  private HashMap<String, String> headers;
+  private HttpRequest request;
+  private MockFileIO mockFileIO;
+
+  protected void setUp() {
+    mockFileIO = new MockFileIO();
+    defaultHandlerRouter = new DefaultHandlerRouter(mockFileIO);
+    customHandlerRouter = new CustomHandlerRouter(defaultHandlerRouter);
```

Comment:
Is this a little weird–that the CustomHandlerRouter needs to work directly with the defaultHandlerRouter to get things going? Anything we can do to get around it?

(I don't have anything specific in mind, just trying to stir up some thoughts).